desktop/cmp/input/CodeInput.ts

```diff
@@ -240,210 +282,227 @@ class CodeInputModel extends HoistInputModel {
     }
 
     @action
     clearSearchResults() {
-        this.cursor = null;
-        this.currentMatchIdx = -1;
-        this.matches.forEach(match => match.textMarker.clear());
         this.matches = [];
+        this.currentMatchIdx = -1;
+        this.updateMatchDecorations();
     }
 
     override destroy() {
-        // Cleanup editor component as per CodeMirror docs.
-        if (this.editor) this.editor.toTextArea();
+        this.editor?.destroy();
         super.destroy();
     }
+
+    //------------------------
+    // Implementation
+    //------------------------
+    private async getExtensionsAsync(): Promise<Extension[]> {
+        const {
+                autoFocus,
+                language,
+                readonly,
+                highlightActiveLine: highlightActiveLineProp,
+                linter: userLinter
+            } = this.componentProps,
+            extensions = [
+                // Theme
+                this.themeCompartment.of(this.getThemeExtension()),
+                // Editor state
+                EditorView.editable.of(!readonly),
+                EditorView.updateListener.of((update: ViewUpdate) => {
+                    if (update.docChanged) this.noteValueChange(update.state.doc.toString());
+                }),
+                // Search & custom highlight
+                search(),
+                syntaxHighlighting(defaultHighlightStyle),
+                highlightSelectionMatches(),
+                this.highlightField,
+                // Editor UI
+                foldGutter(),
+                lineNumbers(),
+                lintGutter(),
+                indentOnInput(),
+                autocompletion(),
+                history(),
+                // Linter
+                userLinter
+                    ? linter(async view => {
+                          const text = view.state.doc.toString();
+                          return await userLinter(text);
+                      })
+                    : [],
+                // Key bindings
+                keymap.of([
+                    ...defaultKeymap,
+                    ...historyKeymap,
+                    ...foldKeymap,
+                    indentWithTab,
+                    {
+                        key: 'Mod-p',
+                        run: () => {
+                            this.onAutoFormat();
+                            return true;
+                        }
+                    }
+                ])
+            ];
+        if (highlightActiveLineProp)
+            extensions.push(highlightActiveLine(), highlightActiveLineGutter());
+        if (autoFocus) extensions.push(this.autofocusExtension);
+        if (language) extensions.push(await this.getLanguageExtensionAsync(language));
+
+        return extensions.filter(it => !isEmpty(it));
+    }
+    private getThemeExtension() {
+        return XH.darkTheme ? dracula : solarizedLight;
+    }
+
+    private async getLanguageExtensionAsync(lang: string) {
```

Comment:
Type this method's output: `Promise<LanguageSupport`. 

Returning `null` instead of `[]` in the unhappy path will fix type errors, and they will be filtered out the same on line 479.